backend/src/v2/compiler/argocompiler/dag.go

```diff
@@ -72,6 +72,15 @@ func (c *workflowCompiler) DAG(name string, componentSpec *pipelinespec.Componen
 		if err != nil {
 			return err
 		}
+		// Add Parallelism Limit if present
+		parallel := int64(kfpTask.GetIteratorPolicy().GetParallelismLimit())
+		if parallel > 0 {
```

Comment:
minor nitpick: You could change the variable name to something like `parallelismLimit` so that it represents a limit on parallelism rather than a count of parallel processes.